### ISSUE TITLE
Running re2c 0.15.3 via src/update produces dirty .c files

### ISSUE THREAD

**Issue by DivineDominion** [fletcher/MultiMarkdown-6#217]
To my surprise, I found that if you run `re3c` v0.15.3 (the version that the .c files indicate), the output is not identical.

Apparently the generated .c  files were at least indented differently than the original re2c output.

This makes suggesting changes weird, because without idempotent outputs, there's no way to highlight actual changes.

@fletcher  It would be great if you could e.g. (a) add instructions to the README how you post-fixed the generated .c files, ie.g. which program you used to change the generated files's indentation; (b) undo custom indentation so the output matches the re2c v0.15.3 default

I also was curious how much re2c v2.x changed, but since not even the outputs using the same version of the tool are identical, diffing isn't helping at all.

**Comment by DivineDominion** [comment]
To ease diffing, I tried `clang-format -i scanners.c` to auto-format the source file, and then I compared the result. 

- It was identical for `scanners.c` using re2c v1.3.0 except for empty newlines.
- It was not identical for `xml.c` using re2c v0.15.3 in other places, unfortunately.

For the sake of making it easier for project contributors to see how much their changes impact, I would suggest running `re2c -i -8 scanners.re > scanners.c` with v1.3.0 again and commit the result unchanged. Not sure about the v0.15.3 based files, yet, though.

@fletcher What do you think?

**Comment by fletcher** [comment]
`make astyle` applies astyle formatting rules (from `.astylerc`) to all source/header files, including the generated parser/lexer files.


As an aside -- You can disagree with my decisions about the formatting choices all you like.  Don't mess with them.  😄

**Comment by DivineDominion** [comment]
@fletcher Thanks for the hint! Overlooked that in the Makefile 😅 

Running `make astyle` on HEAD doesn't mark any file as dirty, so that's a good starting point :) Using v3.1 freshly installed via homebrew.

Here are the updated results for running `re2c` on each file with the appropriate version and then running `make astyle` on the result:

- opml-lexer.c: 9658 lines changed; not just missing whitespace, but also different jump mark labels, e.g.
<img width="496" alt="Screen Shot 2021-07-10 at 21 18 01" src="https://user-images.githubusercontent.com/59080/125174262-502e8e00-e1c4-11eb-8df1-338eeb5c36fa.png">
- itmz-lexer.c: 302994 lines changed, oof
- xml.c: 2858 lines changed; doesn't look tooooo bad, though, but maybe I'm shell-shocked from the 300k lines I had a look at :) xml.c has changes like <img width="536" alt="Screen Shot 2021-07-10 at 21 20 14" src="https://user-images.githubusercontent.com/59080/125174304-9f74be80-e1c4-11eb-87c3-de25251be2b4.png">
- lexer.c:  identical (apart from the comment timestamp) 👍
    - Same differences as mentioned in my last comment; there seems to be a change in the re2c algorithm producing 4177 lines of change
- scanner.c: identical (apart from the comment timestamp) 👍
    - Also identical output with re2c v2.1 👍

The good news is that with `make astyle`, the output of scanner.c and lexer.c is still the same. The bad news is that I don't know why the other three files aren't :)

**Comment by fletcher** [comment]
I generally only update the generated files when I change relevant functionality in some way, so some of the diffs may simply be related to new versions of re2c.   (scanner.c and lexer.c would be the two that were most recently re-generated, I imagine).

The main implications would be if the results of ctest were affected, or if performance was affected.

Probably not worth you spending too much of your time on this.

**Comment by DivineDominion** [comment]
@fletcher The large diffs are the result of using the same version that the .c files topmost comment line say was used when the files were generated. That's surprising me.

I've submitted #218 based on the fact that lexer.re → lexer.c conversion was indeed idempotent with re2c v0.15.3.

Do you think it might be worth updating the generated .c files to at least v1.3.0 to 'pin' a single version for all files at least? Are the `ctest` results the only benchmark to test nothing broke?

**Comment by fletcher** [comment]
Are you accounting for UTF-8 encoding and disabling debug info?

In other words, are you using the update script in the src folder?

**Comment by DivineDominion** [comment]
Yes, I'm using the calls copied and pasted, but instead of "re2c" I write "re2c-0.15.3" or "re2c-1.3.0" :)

(By the way: It appears that nobody else has been using the update script since 2015 or so, since that would've transformed all available .re files to .c files with at least v1.3.0 of re2c, and not just that one scanners.c file; maybe it's time to do that again?)

**Comment by fletcher** [comment]
Pushed newly updated scanner/lexer files.
